commands/env.cmd

```diff
@@ -83,11 +83,29 @@ export TRAEFIK_ADDRESS="$(docker container inspect traefik \
     ' 2>/dev/null || true
 )"
 
+## pause mutagen sync if needed
+if [[ "${WARDEN_PARAMS[0]}" == "stop" ]] \
+    && [[ $OSTYPE =~ ^darwin ]] && [[ -f "${WARDEN_DIR}/environments/${WARDEN_ENV_TYPE}/${WARDEN_ENV_TYPE}.mutagen.yml" ]]
+then
+    warden sync pause
+fi
+
 ## anything not caught above is simply passed through to docker-compose to orchestrate
 docker-compose \
     --project-directory "${WARDEN_ENV_PATH}" -p "${WARDEN_ENV_NAME}" \
     "${DOCKER_COMPOSE_ARGS[@]}" "${WARDEN_PARAMS[@]}" "$@"
 
+## resume mutagen sync if available and php-fpm container id hasn't changed
+if ([[ "${WARDEN_PARAMS[0]}" == "up" ]] || [[ "${WARDEN_PARAMS[0]}" == "start" ]]) \
+    && [[ $OSTYPE =~ ^darwin ]] && [[ -f "${WARDEN_DIR}/environments/${WARDEN_ENV_TYPE}/${WARDEN_ENV_TYPE}.mutagen.yml" ]] \
+    && [[ $(warden sync list | grep -i 'Status: \[Paused\]' | wc -l | awk '{print $1}') == "1" ]] \
+    && [[ $(warden env ps -q php-fpm) ]] \
+    && [[ $(docker container inspect $(warden env ps -q php-fpm) --format '{{ .State.Status }}') = "running" ]] \
+    && [[ $(warden env ps -q php-fpm) = $(warden sync list | grep -i 'URL: docker' | awk -F'/' '{print $3}') ]]
```

Comment:
One thing I am unsure about is if we need to use `=` or `==`. The existing implementation seems to use both and my bash knowledge is limited as to what to prefer here.

Reply:
The short version is `=` is the original equality operator, and at some point bash (and other shells) added `==` as an alias. They are both lexical string comparison operators when enclosed in `[ ]` but this changes when you enclose them in `(( ))` (arithmetic operator) where the two are no longer synonymous.